sound/soc/intel/boards/sof_cirrus_common.c

```diff
@@ -168,11 +168,16 @@ static int cs35l41_compute_codec_conf(void)
 			continue;
 		}
 		physdev = get_device(acpi_get_first_physical_node(adev));
+		acpi_dev_put(adev);
+		if (!physdev) {
+			pr_devel("Cannot find physical node for HID %s UID %u (%s)\n", CS35L41_HID,
+					uid, cs35l41_name_prefixes[uid]);
+			return 0;
```

Comment:
Just curious why you `return 0` here but `continue` when `adev` is not found. Assume it works when less than 4 amps exist, should we `continue` here, too? Is it because that it is not expected that physdev is not found when an `adev` is found?

Reply:
@plbossart recommended the `return 0` and I agree in that if you look at the loop above. Depending on your configuration you might skip `adevs`, but if you have an `adev` but no `physdev` that is a sign you are missing hw that firmware knows about and is an issue.